lib/protoflow-blocks/src/blocks/flow/sort.rs

```diff
@@ -0,0 +1,152 @@
+// This is free and unencumbered software released into the public domain.
+
+use crate::{StdioConfig, StdioError, StdioSystem, System};
+use protoflow_core::{
+    prelude::Vec, types::Any, Block, BlockResult, BlockRuntime, InputPort, Message, OutputPort,
+};
+use protoflow_derive::Block;
+
+/// A block that simply stores all messages it receives.
+///
+/// # Block Diagram
+
+///
+/// # Sequence Diagram
+
+///
+/// # Examples
+///
+/// ## Using the block in a system
+///
+/// ```rust
+/// # use protoflow_blocks::*;
+/// # fn main() {
+/// System::build(|s| {
+///     let stdin = s.read_stdin();
+///     let sort = s.sort();
+///     s.connect(&stdin.output, &sort.input);
+/// });
+/// # }
+/// ```
+///
+/// ## Running the block via the CLI
+///
+/// ```console
+/// $ protoflow execute Sort
+/// ```
+///
+#[derive(Block, Clone)]
+pub struct Sort<T: Message = Any> {
+    /// The input message stream.
+    #[input]
+    pub input: InputPort<T>,
+
+    #[input]
+    pub stop: InputPort<T>,
+
+    /// The output message stream.
+    #[output]
+    pub output: OutputPort<T>,
+
+    /// The internal state storing the messages received.
+    #[state]
+    messages: Vec<T>,
+}
+
+impl<T: Message> Sort<T> {
+    pub fn new(input: InputPort<T>, stop: InputPort<T>, output: OutputPort<T>) -> Self {
+        Self {
+            input,
+            stop,
+            output,
+            messages: Vec::new(),
+        }
+    }
+
+    pub fn messages(&self) -> &Vec<T> {
+        &self.messages
+    }
+}
+
+impl<T: Message + 'static> Sort<T> {
+    pub fn with_system(system: &System) -> Self {
+        use crate::SystemBuilding;
+        Self::new(system.input(), system.input(), system.output())
+    }
+}
+
+impl<T: Message> Block for Sort<T> {
+    fn execute(&mut self, _runtime: &dyn BlockRuntime) -> BlockResult {
+        // if let Some(x) = self.stop.recv()? {
+        //     while let Some(message) = self.input.recv()? {
+        //         if message == x {
+        //             self.messages.sort_by(|x, y| x.partial_cmp(y).unwrap());
+        //             for x in self.messages.iter() {
+        //                 self.output.send(x)?;
+        //             }
+        //             self.messages.clear();
+        //         } else {
+        //             self.messages.push(message);
+        //         }
+        //     }
+        // }
```

Comment:
I'm not sure what the `.unwrap_or()` default value should be when the `partial_cmp` fails as `Ordering::Equal` may not be desired in some cases like:
```rust
let result = f64::NAN.partial_cmp(&1.0);
assert_eq!(result, None);
```

Means the output could have `NAN`'s scattered around, e.g. `vec![0.0, 1.0, f64::NAN, 2.0]` instead of at the front or at the end. Although that's something the `Sort` block's user can work around with a `Filter` block or similar.

But overall to keep true to the one-input-port definition of sort, and [this](https://jpaulm.github.io/fbp/morrison_2005.htm#:~:text=Because%20the%20Sort%20pattern%20does%20not%20start%20to%20output%20any%20data%20until%20all%20the%20incoming%20data%20has%20been%20received):

> Because the Sort pattern does not start to output any data until all the incoming data has been received [...]

May I suggest this?:

```suggestion
        while let Some(message) = self.input.recv()? {
            self.messages.push(message);
        }

        self.messages
            .sort_by(|x, y| x.partial_cmp(y).unwrap_or(Ordering::Equal));

        for message in self.messages.iter() {
            self.output.send(message)?;
        }
```

Reply:
this will depend on how we proceed with the sorting issue